### DISCUSSION TITLE
SDDM - Dynamic Sync Login Screen Wallpaper

### DISCUSSION THREAD

**Discussion by T-Crypt** [prasanthrangan/hyprdots#276] [Show and tell]
I created a script and submitted a merge request to the main dotfiles along with the amdgpu.py and the updated gpuinfo.sh but if you would like to use this before hand and want to be able to change your SDDM login wallpaper feel free to run this script manually. 

(Working on getting this to run in the background without a sudo password -- any ideas welcome) 

SCRIPT: https://github.com/T-Crypt/hyprdots/blob/main/Configs/.config/hypr/scripts/sddmwall.sh

**Comment by prasanthrangan** [comment]
cool idea... I think we can achieve this just by using the symlink `.config/swww/wall.set` for background in `/usr/share/sddm/themes/corners/theme.conf` without a script.

**Reply by T-Crypt** [reply]
I originally tried using the wall.set as I know it has the current background but I was running into issues. 

Let me look at this Monday and maybe we can remove the script and update this. 

SDDM documentation for dynamic/syncing backgrounds was subpar

**Reply by prasanthrangan** [reply]
I just tried it in my system, and it works as expected!

here are the steps -
1. set `Background="/home/<user>/.config/swww/wall.set"` in `/usr/share/sddm/themes/corners/theme.conf`
2. give sddm permissions - `setfacl -m u:sddm:x /home/<user>/.config/swww/wall.set`

```shell
> cd /usr/share/sddm/themes/corners
> grep -i back theme.conf
Background="/home/tittu/.config/swww/wall.set"
> setfacl -m u:sddm:x /home/tittu/.config/swww/wall.set
```

**Reply by T-Crypt** [reply]
For some reason this just results in a white screen for me, doesn't seem to update the image for SDDM.

I'm going to mess with it some more. But I updated permissions for SDDM and updated the theme.conf. This is the error I ran into that I mentioned when I first started the script for SDDM! It doesn't seem to like the wall.set or any of the .config/swww images even with set permissions!

**Reply by prasanthrangan** [reply]
> For some reason this just results in a white screen for me, doesn't seem to update the image for SDDM.

that's definitely due to permission..
maybe try giving `sddm` access to your home dir
```
setfacl -m u:sddm:x /home/<user>/
```
